### ISSUE TITLE
Use a single JSON library

### ISSUE THREAD

**Issue by aalmiray** [in-toto/in-toto-java#100]
At the moment both Jackson and GSON are configured as dependencies. It would be better if there was only one dependency for processing JSON. Hopeful Jackson will be the chosen one.

https://github.com/in-toto/in-toto-java/blob/73afcb46d73cf63f819f10fc591435cdd5c01dd1/pom.xml#L57-L74

**Comment by adityasaky** [comment]
Hi @aalmiray, do you think you could submit a patch converging on one library? We essentially have two implementations under the hood due to a major refactor a while ago and looks like this was a side effect. Thanks!

**Comment by aalmiray** [comment]
Sure. My personal preference would be Jackson as stated before. Does that match the goals of the project?

**Comment by adityasaky** [comment]
cc @SantiagoTorres @Alos

**Comment by Alos** [comment]
When I implemented the new version of the client it was my preference to use Jackson too. However, we did not remove gson for "compatibility" reasons. We could spend some time making sure the gson generated JSON matches the one from Jackson, but I think it requires the addition of more detail unit tests (Especially for the legacy stuff)

**Comment by adityasaky** [comment]
I think switching everything over to jackson is fine. @aalmiray do you want to take a stab at it and working on parts that may have some issues with the switch?

**Comment by aalmiray** [comment]
@adityasaky Yes, I can work on a spike in the coming weeks. I may come back with questions 😅
